[Diagnostics] Add EventPipe Buffer Mode and EventFilter (#54631)
dotnet/docs: docs/core/diagnostics/microsoft-diagnostics-netcore-client.md

        long keywords = 0,

>         IDictionary<string, string> arguments = null)

Author: These aren't compiled here, just here to convey what the signature looks like

    public EventPipeProvider(
        string name,
        EventLevel eventLevel,
        long keywords,
        IDictionary<string, string> arguments,

>         EventPipeProviderEventFilter eventFilter)

Author: These aren't compiled here, just here to convey what the signature looks like

    EventLevel eventLevel,
    long keywords = 0,

>     IDictionary<string, string> arguments = null)

Author: These aren't compiled here, just here to convey what the signature looks like

public EventPipeProvider(
    string name,
    EventLevel eventLevel,
    long keywords,
    IDictionary<string, string> arguments,

>     EventPipeProviderEventFilter eventFilter)

Author: These aren't compiled here, just here to convey what the signature looks like

This may be thrown when the runtime responds with an error to a given command.

### UnsupportedCommandException

```csharp
public class UnsupportedCommandException : ServerErrorException
```

> This may be thrown when the command is not supported by either the library or the target process's runtime.

Reviewer: ```suggestion
This exception is thrown when the command is not supported by either the library or the target process's runtime.
```

### ProfilerAlreadyActiveException

```csharp
public class ProfilerAlreadyActiveException : ServerErrorException
```

This exception is thrown when a profiler is already loaded into the target runtime and another attach is attempted.

### BadEncodingException

```csharp
public class BadEncodingException : ServerErrorException
```

> This is thrown when the target runtime can't decode the command payload and rejects it. From a well-formed client, it usually means the runtime is too old to understand a newer configured option value, so it rejects the request while parsing.

Reviewer: ```suggestion
This exception is thrown when the target runtime can't decode the command payload and rejects it. From a well-formed client, it usually means the runtime is too old to understand a newer configured option value, so it rejects the request while parsing.
```
